sequencer/src/block2.rs

```diff
@@ -117,6 +117,14 @@ impl BlockPayload {
         }
     }
 
+    fn num_namespaces(&self, ns_table_bytes: &[u8]) -> usize {
```

Comment:
We don't need `self` here. This could be an assoc function instead. However... we may wish to keep `self` anyway to avoid an API change in case we want it in the future. Also, it's more consistent with the corresponding `QueryableBlock` trait methods for tx proofs to keep `self`.

But you'll need to write `_self` in ordrer to pacify clippy. Same for `namespace_iter`.

Reply:
ah these methods should also be `pub`